File: app/src/main/java/com/example/carlosyang/myapplication/MainActivity.java
package com.example.carlosyang.myapplication;

import android.app.Activity;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.RelativeLayout;

public class MainActivity extends Activity {
    private RelativeLayout rl_main;
    private Button btn_reset;
    private CustomView customView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        rl_main= (RelativeLayout) findViewById(R.id.rl_main);
        btn_reset= (Button) findViewById(R.id.btn_reset);
        customView=new CustomView(this);
        rl_main.addView(customView);
        btn_reset.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                customView.resetView();
            }
        });
    }
}

File: app/src/main/java/com/example/carlosyang/myapplication/MainActivity.kt
package com.example.carlosyang.myapplication

import android.app.Activity
import android.os.Bundle
import android.widget.Button
import android.widget.RelativeLayout
import org.jetbrains.anko.*

class MainActivity : Activity() {
    private var rl_main: RelativeLayout? = null
    private var btn_reset: Button? = null
    private var customView: CustomView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        relativeLayout {
            backgroundResource = android.R.color.white
            rl_main=this
            customView=CustomView(this@MainActivity).apply {
                addView(this)
            }
            button("重置") {
                btn_reset=this
            }.lparams(width = wrapContent, height = wrapContent) {
                alignParentBottom()
                centerHorizontally()
                bottomMargin = dip(20)
            }.setOnClickListener {
                customView?.resetView()
            }
        }
    }
}
